src/site/topic23.rst

```diff
@@ -189,6 +189,64 @@ Add
+    3. The leftmost node is an interior node
+
+* Finding the minimum is simple
+    * Assuming a root node exists, keep going to the left subtree until there is no more left subtree
+
+* Removing will be a little trickier since we may be removing something that has children that must remain in the tree
+* And further, not only must the children stay in the tree, but the proper binary search tree ordering **must** be preserved
+
+* Consider the following examples and think of how one would need to manage potential children of a node being removed
+    * For this example, assume duplicate values will go to the right
+
+.. image:: img/binarysearchtree_remove0.png
+   :width: 500 px
+   :align: center
+
+.. image:: img/binarysearchtree_remove1.png
+   :width: 500 px
+   :align: center
+
+.. image:: img/binarysearchtree_remove2.png
+   :width: 500 px
+   :align: center
+
+* If we have case 1, where the minimum value is in the root node, this means that
+    * There is no left subtree
+    * There *may* be a right subtree
+    * Therefore, we simply make the root of the right subtree (which may be null) the new root
+
+* For case 1, the ordering is preserved since
+    * Nothing is to the left of the root
+    * Everything to the right of the root is larger than (or equal to) the value in the root
+    * Everything to the right of the right subtree's root must be larger than (or equal to) the right subtree's root
+    * Everything to the left of the right subtree's root must be less than the right subtree's root
+    * Therefore, if all subtrees start with the proper ordering, if the right subtree becomes the new root, the binary search tree's order will be maintained
+
+* If we have case 2, where the minimum is a leaf node, this means that
+    * We have no children to deal with
+    * Therefore, just remove the node
+
+* For case 1, by eliminating only a leaf node, the ordering will not be affected
+
+* If we have case 3, where the minimum is an interior node, this means that
+    * The node has no left subtree --- otherwise the node would not contain the minimum value since the minimum **must** be the leftmost node
+    * A right subtree exists
+    * Therefore, remove the node and make the node's right subtree's root replace it
+        * The node being removed's parent's left child will become the node being removed's right child
+
+* For case 3, the ordering is preserved in the same way as case 1
+    * Nothing is to the left of the node
+    * Everything to the right of the node to be removed is larger than (or equal to) the value in the node
+    * Everything to the right of the node's right subtree's root must be larger than (or equal to) the right subtree's root
+    * Everything to the left of the right subtree's root must be less than the right subtree's root
+    * Therefore, if all subtrees start with the proper ordering, if the right subtree replaces the node being removed, the binary search tree's order will be maintained
+
```

Comment:
There's opportunity to show that the three cases here are actually all identical. You replace a node with the node's right subtree in all of them.